### ISSUE TITLE
Description and default interval value issues with SolidFire snapshot schedule module

### ISSUE THREAD

**Issue by scaleoutsean** [ansible-collections/netapp#44]
##### SUMMARY

1) The module description is completely wrong (maybe not just this module).
It says it's for [account management ](https://github.com/ansible-collections/netapp/blob/ba448fe9dbbe7e9c53493a0114f66f410659dd91/ansible_collections/netapp/elementsw/plugins/modules/na_elementsw_snapshot_schedule.py#L26).

2) There's an unnecessary [default non-zero value](https://github.com/ansible-collections/netapp/blob/ba448fe9dbbe7e9c53493a0114f66f410659dd91/ansible_collections/netapp/elementsw/plugins/modules/na_elementsw_snapshot_schedule.py#L213) for `time_interval_days`. 

```
    schedule_type:
        description:
        - Schedule type for creating schedule.
        choices: ['DaysOfWeekFrequency','DaysOfMonthFrequency','TimeIntervalFrequency']

    time_interval_days:
        description: Time interval in days.
        default: 1
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

* [na_elementsw_snapshot_schedule.py](https://github.com/ansible-collections/netapp/blob/ba448fe9dbbe7e9c53493a0114f66f410659dd91/ansible_collections/netapp/elementsw/plugins/modules/na_elementsw_snapshot_schedule.py)

##### ANSIBLE VERSION

```shell
ansible 2.9.10
  config file = None
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/vagrant/.local/lib/python3.6/site-packages/ansible
  executable location = /home/vagrant/.local/bin/ansible
  python version = 3.6.9 (default, Apr 18 2020, 01:56:04) [GCC 8.4.0]
```

##### CONFIGURATION

* Element OS 11.7

##### EXPECTED RESULTS

* The module description correctly describes its purpose
* The module does not use unreasonable default values

##### ACTUAL RESULTS

* The module description is wrong
* At least one variable has an unreasonable defaults value (`time_interval_days=1`). Actually I don't even understand why `time_interval_days` is necessary in the first place (there's DaysOfWeekFrequency and DaysOfMonthFrequency as well as `time_interval_hours=24`) but if it can't be removed, then it should have a reasonable default value like the other two time intervals, namely `0`.

**Comment by carchi8py** [comment]
for the first issue i have story 2900 open internally to fix that.

**Comment by carchi8py** [comment]
For the second issue i'm talking with the team to see if there was a reason we set this to 1 vs 0.

**Comment by lonico** [comment]
account shows only once in the module.  Needs to changed to snapshot schedule.

The parameters  strictly map the SolidFire SDK.  I wonder if we could provide a better abstraction.  But as it is, it just collects the value the SDK expects.

I'm generally against using defaults, as I'd rather use the backend defaults implicitly.   But whats' so wrong about using a default of 1.  It can easily be changed.

**Comment by scaleoutsean** [comment]
@lonico I did check the API Reference Guide before I submitted the bug and didn't see it (and that was one of the reasons that prompted me to go ahead with this issue submission)

```
{
    "method": "CreateSchedule",
    "params": {
        "attributes": {
            "frequency": "<One of: Time Interval, Days of Week or Days of Month>"
        },
        "hours": <Optional Hours between snapshots, used for Time Interval mode>,
        "minutes": <Optional Minutes between snapshots, used for Time Interval mode>,
        "paused": <Optional Boolean true or false>,
        "recurring": <Optional Boolean true or false>,
        "scheduleName": "<Name for the schedule>",
        "scheduleType": "snapshot",
        "scheduleInfo": {
            "volumeID": <Volume ID to be snapped>,
            "volumes": [<CSV List of Volume IDs to be group snapped>],
            "name": "<Name of the snapshot>",
            "enableRemoteReplication": <Boolean true or false>
            "retention": "<Optional amount of time to retain the snapshot in HH:mm:ss>"
        },
        "startingDate": "<Optional Starting Date in ISO 8601 Format>",
        "monthdays": [<Optional CSV List of days to run>],
        "weekdays": [
            {"day": <Day Number>, "offset": 1},
            {"day": <Day Number>, "offset": 1}
        ]
    },
    "id": 1
}
```
I don't see it.

Maybe "a" larger than zero value needs to be provided for one of the parameters, but since it can be 0 (assuming some other interval parameter is >0), it could be zero and errors could be returned as they are by the API.

**Comment by lonico** [comment]
we're getting it from from solidfire.custom.models import TimeIntervalFrequency

But yes, I'd rather let the API error out, if it errors out cleanly.  Or maybe checking we don't have (0, 0, 0) would be enough.

**Comment by lonico** [comment]
And the default is 0 so why change it ?

```
class TimeIntervalFrequency(Frequency):
    """
    Use this class when setting the frequency of a Schedule to Time Interval.
    Time Interval schedules trigger after the specified interval of time has
    passed.

    :param minutes:  The minutes that will elapse before the next snapshot is
    created.
    Valid values are: 0 - 59. Default is 0.
    :type minutes: int

    :param hours: The hours that will elapse before the next snapshot is
    created.
    Valid values  0 - 24. Default is 0.
    :type hours: int

    :param days: The days that will elapse before the next snapshot is created.
    Default is 0.
    :type days: int
    """
    minutes = data_model.property(
        "minutes", int,
        array=False, optional=True,
        documentation="Number of minutes between snapshots. "
                      "Valid values are: 0 - 59"
    )

    hours = data_model.property(
        "hours", int,
        array=False, optional=True,
        documentation="Number of hours between snapshots. "
                      "Valid values are: 0 - 24"
    )
    days = data_model.property(
        "days", int,
        array=False, optional=False,
        documentation="Number of days between snapshots."
    )
```

**Comment by lonico** [comment]
We'll remove all defaults.  Interestingly, the API can create a snapshot schedule with a value of (0, 0, 0) for the 3 intervals.  I guess it does nothing, but we'll prevent that.

**Comment by scaleoutsean** [comment]
Good wrt the removal..

>  I guess it does nothing, but we'll prevent that.

As I said in the first comment, I can choose DaysOfWeekFrequency and DaysOfMonthFrequency. I don't think I actually need it.

Or I don't have to start it, I can just pre-create it for late and leave the schedule parked there.

```
{
    "method": "CreateSchedule",
    "params": {
        "attributes": {
            "frequency": "Time Interval"
        },
        "hours": 0,
        "paused": true,
        "recurring": false,
        "scheduleName": "paused",
        "scheduleType": "snapshot",
        "startingDate": "2030-01-01T00:30",
        "scheduleInfo": {
            "volumeID": 3,
            "name": "nobug",
            "enableRemoteReplication": false,
            "retention": "02:00:00" }
        
    },
    "id": 1
}
```

**Comment by lonico** [comment]
fixed with 506f67fa601c8361dd11a793970ff515b0a38aeb
